src/game/combat.cpp

```diff
@@ -57,6 +58,10 @@ void Combat::update(float dt) {
         updateAI();
     }
     updateRounds(dt);
+
+    if (!wasActive && isActive()) onEnterCombatMode();
```

Comment:
nit: I don't like this construct, but can't quite put why. Let me know what you think of this variant:

```cpp
enum class CameraStyleType {
    Default,
    Combat
};

class Area {
public:
    void setCameraStyleType(CameraStyleType type) {
        if (_cameraStyleType == type) return;

        const CameraStyle &style = type == CameraStyleType::Combat ? _camStyleCombat : _camStyleDefault;
        _thirdPartyCamera->setStyle(style);
    }

private:
    CameraStyleType _cameraStyleType { CameraStyleType::Default };
    CameraStyle _camStyleDefault;
    CameraStyle _camStyleCombat;
};

class Combat {
public:
    void update(float dt) {
        CameraStyleType type = isActive() ? CameraStyleType::Combat : CameraStyleType::Default;
        _game->module()->area()->setCameraStyleType(type);
    }
};
```

Reply:
I like using a enum to represent CameraStyleType, but I think setting camera style every frame is kind of unnecessary.

We'll probably need an `onEnterCombatMode` at some point for auto-pausing, don't we?

Reply:
Camera _style_ is not being set every frame here. There is a single function call and a conditional statement per frame - zero overhead, basically.

Reply:
I see what you mean. My initial thoughts were that since there is no combat-related logic inside `onEnterCombatMode`, there is no reason to track combat activation inside Combat.

Reply:
How about this then?

```cpp
class Combat {
public:
    bool isActive() const { return _active; } 

private:
    bool _active { false };

    void updateActivation() {
        bool active = !_combatants.empty();
        if (_active == active) return;

        if (active) onEnterCombatMode();
        else onExitCombatMode();

        _active = active;
    }
};
```

Reply:
Yeah I like this one, thanks